contracts/Zauction.sol

```diff
@@ -76,27 +92,109 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     uint256 expireBlock
+  ) external {
+    require(startBlock <= block.number, "zAuction: auction hasn't started");
+    require(expireBlock > block.number, "zAuction: auction expired");
+    require(minbid <= bid, "zAuction: cannot accept bid below min");
+    require(bidder != msg.sender, "zAuction: cannot sell to self");
+    require(msg.sender == hub.ownerOf(domainTokenId), "Only Owner");
+
+    IRegistrar domainRegistrar = hub.getRegistrarForDomain(domainTokenId);
+
+    // Don't include a token address when recreating bid data for legacy bids
+    bytes32 data = createBid(
+      bidNonce,
+      bid,
+      domainTokenId,
+      minbid,
+      startBlock,
+      expireBlock,
+      address(0)
+    );
+
+    require(
+      bidder == recover(toEthSignedMessageHash(data), signature),
+      "zAuction: recovered incorrect bidder"
+    );
+    require(!consumed[bidder][bidNonce], "zAuction: data already consumed");
+
+    consumed[bidder][bidNonce] = true;
+
+    uint256 topLevelDomainId = getTopLevelIdWithUpdate(domainTokenId);
+    // Transfer payment, royalty to minter, and fee to topLevel domain
+    paymentTransfers(
+      bidder,
+      bid,
+      msg.sender,
+      topLevelDomainId,
+      domainTokenId,
+      wildToken
+    );
+
+    // Owner -> Bidder, send NFT
+    domainRegistrar.safeTransferFrom(msg.sender, bidder, domainTokenId);
+
+    emit BidAccepted(
+      bidNonce,
+      bidder,
+      msg.sender,
+      bid,
+      address(domainRegistrar),
+      domainTokenId,
+      expireBlock,
+      address(wildToken),
+      topLevelDomainId
+    );
+  }
+
+  // Accept a bid but with the addition of a bidToken parameter to support v2.1 changes
+  /// @param signature type encoded message signed by the bidder
+  /// @param bidNonce unique per address auction identifier chosen by seller
+  /// @param bidder address of who the seller says the bidder is, for confirmation of the recovered bidder
+  /// @param bid token amount bid
+  /// @param tokenId token id we are transferring
+  /// @param minbid minimum bid allowed
+  /// @param startBlock block number at which acceptBid starts working
+  /// @param expireBlock block number at which acceptBid stops working
+  /// @param bidToken the token used in payment for the bid
+  function acceptBidV2(
+    bytes memory signature,
+    uint256 bidNonce,
+    address bidder,
+    uint256 bid,
+    uint256 tokenId,
+    uint256 minbid,
+    uint256 startBlock,
+    uint256 expireBlock,
+    IERC20 bidToken
   ) external {
     require(startBlock <= block.number, "zAuction: auction hasn't started");
     require(expireBlock > block.number, "zAuction: auction expired");
     require(minbid <= bid, "zAuction: cannot accept bid below min");
     require(bidder != msg.sender, "zAuction: cannot sell to self");
     require(msg.sender == hub.ownerOf(tokenId), "Only Owner");
 
+    // Disallow accepting bid if made in the wrong ERC20 token
+    IERC20 domainToken = getTokenForDomain(tokenId);
```

Comment:
Nit: possibly misleading name since I might think that this is the NFT (which is a domain, which is a token)
Could you name this something like `paymentToken`